runtime/server/downloads.go

```diff
@@ -0,0 +1,92 @@
+package server
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/queries"
+	"google.golang.org/protobuf/proto"
+)
+
+type DownloadHandler struct {
+	Runtime *runtime.Runtime
+}
+
+func (h *DownloadHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
```

Comment:
Can just implement it on the server like the other handlers, like `func (s *Server) downloadHandler(w http.ResponseWriter, r *http.Request)`, and use `http.HandlerFunc` to mount it on the `mux`